internal/api/impl/auth/oauth.go

```diff
@@ -253,12 +253,12 @@ func (e *oAuthEndpoint) readCodeVerifierCookie(ctx echo.Context) (state string,
 
 func (e *oAuthEndpoint) CollectRoutes(g *route.Group) {
 	// Add routes for the "Authorization Code" flow
-	g.GET(fmt.Sprintf("/%s/%s/%s", utils.AuthKindOAuth, e.slugID, utils.PathLogin), e.authHandler, true)
-	g.GET(fmt.Sprintf("/%s/%s/%s", utils.AuthKindOAuth, e.slugID, utils.PathCallback), e.codeExchangeHandler, true)
+	g.GET(fmt.Sprintf("/%s/%s/%s", utils.AuthKindOAuth, e.slugID, utils.PathLogin), withOAuthErrorMdw(e.authHandler), true)
```

Comment:
with the PR #1838 we could do something like that:

```suggestion
oauthGroup := g.Group(fmt.Sprintf("/%s/%s", utils.AuthKindOAuth, e.slugID), withOauthErrorMDW)
oauthGroup.Get(fmt.Sprintf("/%s", utils.PathLogin, true)
oauthGroup.Get(fmt.Sprintf("/%s", utils.PathCallback, true)

oauthGroup.Post(fmt.Sprintf("/%s", utils.PathDeviceCode, true)
oauthGroup.Post(fmt.Sprintf("/%s", utils.PathToken, true)
```